src/Model/Behavior/SearchBehavior.php

```diff
@@ -111,6 +111,6 @@ protected function _getAllFilters($collection = null)
             $manager = $this->searchManager();
         }
 
-        return !is_null($collection) ? $manager->getFilters($collection) : $manager->all();
+        return ($collection !== null) ? $manager->getFilters($collection) : $manager->all();
```

Comment:
Makes me wonder why `is_null()` was introduced to PHP then.
Thanks for the cleanups